pyresample/bilinear/__init__.py

```diff
@@ -210,97 +203,448 @@ def get_bil_info(source_geo_def, target_area_def, radius=50e3, neighbours=32,
+        """Load bilinear resampling look-up tables and initialize the resampler."""
+        raise NotImplementedError
+
+
+def _check_fill_value(fill_value, dtype):
+    """Check that fill value is usable for the data."""
+    if fill_value is None:
+        if np.issubdtype(dtype, np.integer):
+            fill_value = 0
+        else:
+            fill_value = np.nan
+    elif np.issubdtype(dtype, np.integer):
+        if np.isnan(fill_value):
+            fill_value = 0
+        elif np.issubdtype(type(fill_value), np.floating):
+            fill_value = int(fill_value)
+
+    return fill_value
+
+
+def _get_output_xy(target_geo_def):
+    out_x, out_y = target_geo_def.get_proj_coords()
+    return np.ravel(out_x),  np.ravel(out_y)
+
+
+def _get_input_xy(source_geo_def, proj, valid_input_index, index_array):
+    """Get x/y coordinates for the input area and reduce the data."""
+    return proj(
+        *array_slice_for_multiple_arrays(
+            index_array,
+            array_slice_for_multiple_arrays(
+                valid_input_index,
+                mask_coordinates(*_get_raveled_lonlats(source_geo_def))
+            )
+        )
+    )
+
+
+def array_slice_for_multiple_arrays(idxs, data):
+    """Slices multiple arrays using the same indices."""
+    return [d[idxs] for d in data]
+
+
+def mask_coordinates(lons, lats):
+    """Mask invalid coordinate values."""
+    idxs = (find_indices_outside_min_and_max(lons, -180., 180.) |
+            find_indices_outside_min_and_max(lats, -90., 90.))
+    return _np_where_for_multiple_arrays(idxs, (np.nan, np.nan), (lons, lats))
+
+
+def find_indices_outside_min_and_max(data, min_val, max_val):
+    """Return array indices outside the given minimum and maximum values."""
+    return (data < min_val) | (data > max_val)
+
+
+def _get_raveled_lonlats(geo_def):
+    lons, lats = geo_def.get_lonlats()
+    if lons.size == 0 or lats.size == 0:
+        raise ValueError('Cannot resample empty data set')
+    elif lons.size != lats.size or lons.shape != lats.shape:
+        raise ValueError('Mismatch between lons and lats')
+
+    return np.ravel(lons), np.ravel(lats)
+
+
+def _get_bounding_corners(in_x, in_y, out_x, out_y, neighbours, index_array):
+    """Get bounding corners.
+
+    Get four closest locations from (in_x, in_y) so that they form a
+    bounding rectangle around the requested location given by (out_x,
+    out_y).
+
+    """
+    # Find four closest pixels around the target location
+
+    stride, valid_corners = _get_stride_and_valid_corner_indices(
+        out_x, out_y, in_x, in_y, neighbours)
+    res = []
+    indices = []
+    for valid in valid_corners:
+        x__, y__, idx = _get_corner(stride, valid, in_x, in_y, index_array)
+        res.append(np.transpose(np.vstack((x__, y__))))
+        indices.append(idx)
+    res.append(np.transpose(np.vstack(indices)))
+
+    return res
+
+
+def _get_fractional_distances(pt_1, pt_2, pt_3, pt_4, out_x, out_y):
```

Comment:
That's a lot of arguments, could you maybe put them together? Maybe introducing an object for the 4 points would help? https://refactoring.guru/introduce-parameter-object

Reply:
I went the "easy" way and used tuples.